### DISCUSSION TITLE
can a snippet access it's context?

### DISCUSSION THREAD

**Discussion by ghistes** [L3MON4D3/LuaSnip#1222] [Q&A]
Hi,

I wonder if it possible for a snippet to access it's context (i.e. lines preceding it), for example I would like to have a function node compute a value based on something that appears in the buffer a few lines above from the line where I triggered it (I tried to use a multiline regex-trigger but could not get it to work).

The value I would be interested in is the nearest match for a given regex - but it could appear anywhere between the start of the buffer and the place where I trigger the snippet.

If that is understandable at all - is there a way to do it with luasnip?

Many thanks.

**Comment by bew** [comment]
Take a look at this tutorial that does exactly that at the end:
https://youtube.com/watch?v=KtQZRAkgLqo&t=1350s
